### PR TITLE
refactor: transform react-native => ts & modify jestConfig: delete th…

### PR DESCRIPTION
**Pull request by myNameIsDu**
refactor: transform react-native => ts & modify jestConfig: delete the JS matching pattern

link #1737 
 
there is a problem:

image
In this case, I've deleted the previous argument

### CODE REVIEW

File: test/react-native/batch-integration.tsx [hunk 1]
```diff
@@ -467,33 +570,35 @@ describe('React Native', () => {
...
         const button = rendered.getByTestId(testID)
         rtl.fireEvent.press(button)
       }
 
-      const clickAndRender = (rendered, testID) => {
+      const clickAndRender = (rendered: RenderedType) => {
         // Note: Normally we'd wrap this all in act(), but that automatically
         // wraps your code in batchedUpdates(). The point of this bug is that it
         // specifically occurs when you are _not_ batching updates!
         clickButton(rendered, 'setTimeout')
```
Comment (markerikson) [inline test/react-native/batch-integration.tsx:591]: :hand: Yeah, I think there's actually a bug in this test here. It should be `clickButton(rendered, testID)` - I apparently missed passing that through when I was writing this :)

Can you restore the arguments for this section and fix this line?

**Review by markerikson — CHANGES_REQUESTED** [review]

File: test/react-native/batch-integration.tsx (hunk 1 above)
Comment (myNameIsDu) [reply test/react-native/batch-integration.tsx:591]: Okay, no problem

Comment (myNameIsDu) [reply test/react-native/batch-integration.tsx:591]: Okay, no problem

**Comment by markerikson** [comment]
Looks good, thanks!

**Pull request merged**